PAC_SET_HEADER_LIB_PATH

raffenet opened the pull request:
1. Remove warning/scrubbing of valid options.
2. Improves hcoll configure options.

raffenet commented:
test:jenkins/ch3/most

raffenet commented:
test:jenkins/ch3/most

raffenet commented:
Fixed one more dumb mistake.
test:jenkins/ch3/most

raffenet commented:
I seem to have confused Jenkins.
test:jenkins/ch3/mxm

raffenet commented:
@pavanbalaji this is ready for review when you have a chance.

pavanbalaji commented on src/mpid/common/hcoll/subconfigure.m4:
```diff
@@ -1,19 +1,14 @@
 	PAC_SET_HEADER_LIB_PATH(hcoll)
 	PAC_PUSH_FLAG(LIBS)
-	if test "$enable_hcoll" != "no"; then
+	if test ! -z "$with_hcoll" -a "$with_hcoll" != "no"; then
```
Why does `$with_hcoll` need to be not-NULL?  What is it's already present in the system path?  I should be able to simply pass `--with-hcoll` to enable it, isn't it?

raffenet replied:
Yes, that sounds like a better test. I'll revise.

raffenet replied:
Oh, I think what I was trying to accomplish was to make sure configure would fail if the user passed `--with-hcoll=[yes|path/to/install]`. If the user didn't supply `--with-hcoll` at all, then configure should just disable it an move on. I think my logic was still a bit wrong and will revise in any case.

raffenet commented:
@pavanbalaji this was fresh in my mind so I rebased and split into a couple of commits. Basically, this removes the enable/disable option that shouldn't have been added, and adds a warning when auto-detection fails.

pavanbalaji commented:
This is relying on `with_hcoll` being set.  That might not be the case if the user uses `--with-hcoll-lib`, for example.

raffenet commented:
That's true. Do you think we should add something like below to the warning condition? If none of those conditions are met it means the user supplied something and we should throw an error.
```
-o -z "$with_hcoll_lib" -o -z "$with_hcoll_include"
```

pavanbalaji commented:
I'd say set `have_hcoll` to `no`, when `with_hcoll` is `no`, and have all following logic simply check for `have_hcoll` = `yes`.  That way we are covering both cases, i.e., where the user provided a path as well as the system path being valid for hcoll.

raffenet commented:
Sorry for the long delay @pavanbalaji. I have updated the PR based on your suggestion, and added another commit to confdb to make the `with/without` help text a little clearer. Please review when you have a chance.

pavanbalaji commented:
Was this tested?

raffenet commented:
Oops, only locally. I'll give it a sanity check before any merge.

raffenet commented:
I'm also reminded I need to add a simplified `hcoll` test back to Jenkins. The full MPICH testsuite runs into issues because of the lack of non-contig support at the moment. That is being worked on in a separate PR.
test:jenkins/ch3/most

raffenet commented:
test:jenkins/ch4/most

raffenet commented:
Tests look good.
